packages/luna/src/components/Lessons/BookLesson/BookLessonDialog.tsx

```diff
@@ -156,80 +163,40 @@ export const BookLessonDialog: React.FC<{
     return { min, max };
   }, []);
 
-  const unpackedRules = useMemo(() => {
-    return unpackRules({
-      rules,
-      slots,
-      /**
-       * ! update {@link unpackRules} logic and make `start` date inclusive.
-       */
-      start: dateBounds.min.subtract(1, "day").toISOString(),
-      end: dateBounds.max.add(1, "day").toISOString(),
-    });
-  }, [dateBounds.max, dateBounds.min, rules, slots]);
+  const unBookedSlots = useMemo(
+    () => subtractSlots({ slots, subslots: bookedSlots }),
+    [slots, bookedSlots]
+  );
 
   const selectDaySlots = useCallback(
     (day: Dayjs) => {
-      const daySlots = unpackedRules.filter(
+      const daySlots = unBookedSlots.filter(
         (event) =>
           day.isSame(event.start, "day") || day.isSame(event.end, "day")
       );
 
-      const unpackedSlots: AttributedSlot[] = flattenDeep(
-        daySlots.map((rule) => splitRuleEvent(rule, duration))
-      ).map((event) => ({
-        ruleId: event.id,
-        slotId: event.id,
-        start: event.start,
-        end: event.end,
-        bookable: dayjs(event.start).isAfter(
-          dayjs().add(notice || 0, "minutes")
-        ),
-      }));
-
-      return unpackedSlots;
+      return getSubSlots(daySlots, duration);
```

Comment:
I believe that the `bookable` flag is dropped. This mean that the booked slots will not be rendered.